### ISSUE TITLE
[Feature Request] support for relative paths

### ISSUE THREAD

**Issue by Heus-Sueh** [Alexays/Waybar#3715]
The need to put the absolute path becomes unbearable when you include several modules:
```
  "include": [
    "~/.config/waybar/modules/hyprland.jsonc",
    "~/.config/waybar/modules/battery.jsonc",
    "~/.config/waybar/modules/backlight.jsonc",
    "~/.config/waybar/modules/custom-music.jsonc",
    "~/.config/waybar/modules/custom-colorpicker.jsonc",
	...
  ],
```

I believe that the waybar could easily assume the path based on where the `config.jsonc` is being loaded and thus making it easier.
```
  "include": [
	# the waybar would be assuming that the user is referencing the modules folder which is located in the same place as config.json
    "modules/hyprland.jsonc", # absolute path (~/.config/waybar/modules/hyprland.jsonc) 
    "modules/battery.jsonc",
    "modules/backlight.jsonc",
    "modules/custom-music.jsonc",
    "custom-colorpicker.jsonc",
	...
  ],
```

**Comment by RobertMueller2** [comment]
This seems to work now, can you test again with newer version?

**Comment by RobertMueller2** [comment]
no, i see what you mean now, still doesn't work depending on where you start waybar from. However, what can be done is to use wildcards now, maybe that helps?

**Comment by Heus-Sueh** [comment]
> no, i see what you mean now, still doesn't work depending on where you start waybar from. However, what can be done is to use wildcards now, maybe that helps?

How could this be done? It seems interesting, but what if you don't want to include a module?

**Comment by RobertMueller2** [comment]
You could rename modules you're not using so they're not matched. If you use several configs with the same set of modules, you could use a config.d directory and then conf1-config.d, conf2-config.d and symlink the desired modules from config.d to the other directories.

Another thing you could try to achieve your original idea is to start waybar from a script and change the working directory, e.g.
```
#!/bin/sh

cd $HOME/.config/waybar
exec waybar "$@"
```

**Comment by lethedata** [comment]
For anyone looking into this and running through a systemd unit, you can create a drop-in snippet that sets the WorkingDirectory to where your configs are.

`systemctl --user edit waybar.service` 

```ini
# %E = $XDG_CONFIG_HOME
[Service]
WorkingDirectory=%E/waybar
```

**Comment by Alexays** [comment]
Relative `include` paths do work — they're just resolved against Waybar's **current working directory**, not the config file's location (see `findIncludePath` in `src/config.cpp`). So the clean fix is to make Waybar start in your config dir.

If you launch via the systemd user service, drop in a working directory (from @lethedata in-thread):

```ini
# systemctl --user edit waybar.service
[Service]
WorkingDirectory=%E/waybar   # %E = $XDG_CONFIG_HOME
```

...then `"include": ["modules/battery.jsonc"]` resolves to `~/.config/waybar/modules/battery.jsonc`. If you start it manually, a tiny wrapper does the same:

```sh
#!/bin/sh
cd "$HOME/.config/waybar" && exec waybar "$@"
```

Glob/wildcard includes are also supported now. Closing since the workflow is covered, but reopen if that doesn't fit your setup. 🙏
